gobblin-runtime/src/main/java/gobblin/runtime/app/ServiceBasedAppLauncher.java

```diff
@@ -224,13 +226,30 @@ private void addJobExecutionServerAndAdminUI(Properties properties) {
 
       if (adminUiServerEnabled) {
         LOG.info("Will launch the admin UI server");
-        addService(new AdminWebServer(properties, executionInfoServer.getAdvertisedServerUri()));
+        addService(createAdminServer(properties, executionInfoServer.getAdvertisedServerUri()));
       }
     } else if (adminUiServerEnabled) {
       LOG.warn("Not launching the admin UI because the job execution info server is not enabled");
     }
   }
 
+  public static Service createAdminServer(Properties properties,
+                                          URI executionInfoServerURI) {
+    String factoryClassName = properties.getProperty(ConfigurationKeys.ADMIN_SERVER_FACTORY_CLASS_KEY,
```

Comment:
On a side note, in a Spring app this could be a lot cleaner as it would be able to be autoconfigured and conditional on the class being in the classpath.